claims/time.go

```diff
@@ -0,0 +1,66 @@
+package claims
+
+import (
+	"fmt"
+	"time"
+)
+
+// TimeSource defines a source for the current time
+type TimeSource func() time.Time
+
+func defaultTimeSource() time.Time {
+	return time.Now()
+}
+
+// Time handles time related registered claim verification, using an optional
+// TimeSource as point of reference
+type Time struct {
+	T TimeSource
```

Comment:
Also, use a more descriptive name here. Stuff like `t.T` from below is kinda confusing otherwise.